demos/sample_azure_iot/sample_azure_iot.c

```diff
@@ -353,145 +356,154 @@ static void prvAzureDemoTask( void * pvParameters )
-                                              sampleazureiotMESSAGE, lPublishCount );
-            xResult = AzureIoTHubClient_SendTelemetry( &xAzureIoTHubClient,
-                                                       ucScratchBuffer, ulScratchBufferLength,
-                                                       &xPropertyBag, eAzureIoTHubMessageQoS1, NULL );
+            /* Sending a default property (Content-Encoding). */
+            xResult = AzureIoTMessage_PropertiesAppend( &xPropertyBag,
+                                                        ( uint8_t * ) AZ_IOT_MESSAGE_PROPERTIES_CONTENT_ENCODING, sizeof( AZ_IOT_MESSAGE_PROPERTIES_CONTENT_ENCODING ) - 1,
+                                                        ( uint8_t * ) sampleazureiotMESSAGE_CONTENT_ENCODING, sizeof( sampleazureiotMESSAGE_CONTENT_ENCODING ) - 1 );
             configASSERT( xResult == eAzureIoTSuccess );
 
-            LogInfo( ( "Attempt to receive publish message from IoT Hub.\r\n" ) );
-            xResult = AzureIoTHubClient_ProcessLoop( &xAzureIoTHubClient,
-                                                     sampleazureiotPROCESS_LOOP_TIMEOUT_MS );
+            /* How to send an user-defined custom property. */
+            xResult = AzureIoTMessage_PropertiesAppend( &xPropertyBag, ( uint8_t * ) "name", sizeof( "name" ) - 1,
+                                                        ( uint8_t * ) "value", sizeof( "value" ) - 1 );
             configASSERT( xResult == eAzureIoTSuccess );
 
-            if( lPublishCount % 2 == 0 )
+            /* Publish messages with QoS1, send and process Keep alive messages. */
+            for( lPublishCount = 0;
+                 lPublishCount < lMaxPublishCount && xAzureSample_IsConnectedToInternet();
+                 lPublishCount++ )
             {
-                /* Send reported property every other cycle */
                 ulScratchBufferLength = snprintf( ( char * ) ucScratchBuffer, sizeof( ucScratchBuffer ),
-                                                  sampleazureiotPROPERTY, lPublishCount / 2 + 1 );
-                xResult = AzureIoTHubClient_SendPropertiesReported( &xAzureIoTHubClient,
-                                                                    ucScratchBuffer, ulScratchBufferLength,
-                                                                    NULL );
+                                                  sampleazureiotMESSAGE, lPublishCount );
+                xResult = AzureIoTHubClient_SendTelemetry( &xAzureIoTHubClient,
+                                                           ucScratchBuffer, ulScratchBufferLength,
+                                                           &xPropertyBag, eAzureIoTHubMessageQoS1, NULL );
+                configASSERT( xResult == eAzureIoTSuccess );
+
+                LogInfo( ( "Attempt to receive publish message from IoT Hub.\r\n" ) );
+                xResult = AzureIoTHubClient_ProcessLoop( &xAzureIoTHubClient,
```

Comment:
If we lose connectivity during one of these steps, will that cause the configAssert to crash the sample instead of getting caught by the xAzureSample_IsConnectedToInternet in the loop check?

Reply:
Yes it will. The internet connection checks are for now just for some key points of the code. 
More test will be done later to try to extend the check to all potential I/O failure points.